src/v/config/configuration.cc

```diff
@@ -827,6 +827,25 @@ configuration::configuration()
       "remote storage (sec)",
       required::no,
       std::nullopt)
+  , cloud_storage_cache_directory(
+      *this,
+      "cloud_storage_cache_directory",
+      "Directory for archival cache. Should be present when "
+      "`cloud_storage_enabled` is present",
+      required::no,
+      std::nullopt)
```

Comment:
should we force the cache directory to exist in redpanda's data directory for now? that would eliminate the chance of conflict, make configuration easier, and could be an optimization that we add later. in either case, using a real default is probably useful for making configuration simpler.

Reply:
> should we force the cache directory to exist in redpanda's data directory for now?

I am in favor of this TBH

Reply:
In case of using real default I think we want to use `data_dir`, but the data dir value is not available here yet. We can handle that when archival service will parse configs and pass the `cache_dir` to the `cache` constructor.

Reply:
you could delete the configuration option and then in the cache use the directory `config().data_directory() + /archival_cache` or something like that.